web-2.5/frontend/src/App.tsx

```diff
@@ -0,0 +1,105 @@
+import type { FormEventHandler } from 'react';
+import { useState } from 'react';
+import './App.css';
+import mw from 'mw';
+
+function App() {
+  const [name, setName] = useState('');
+  const [experience, setExperience] = useState(0);
+  const [wallet, setWallet] = useState('');
+  const [renamePending, setRenamePending] = useState(false);
+
+  const [newName, setNewName] = useState('');
+
+  const fetchPlayer = async (userWallet: string) => {
+    const response = await mw.getUserState(userWallet);
+    if (!response.success) {
+      console.error('Failed to get user data', response.errorMessage, response.errorCode);
+    } else {
+      setExperience(response.result.experience);
+      setName(response.result.name);
+    }
+  };
+
+  const characterRename: FormEventHandler = async e => {
+    e.preventDefault();
+
+    setRenamePending(true);
+    const response = await mw.renamePlayer(newName);
+    if (response.success) {
+      setRenamePending(false);
+      setName(newName);
+    } else {
+      console.log('Failed to rename player');
+      fetchPlayer(wallet);
+    }
+  };
+
+  const userWalletLogin = async () => {
+    const response = await mw.userWalletLogin('metamask');
+    if (!response.success) {
+      console.log('Error while logging in address:', response.errorMessage, response.errorCode);
+    } else {
+      const { walletAddress } = response.result;
+      console.log('Successfully logged in address:', walletAddress);
+      setWallet(walletAddress);
+      fetchPlayer(walletAddress);
+    }
+  };
+
+  return (
+    <div className="container">
+      <header>
+        <h1>
+          Web 2.5 Example{' '}
+          {name && (
+            <span>
+              {name} ({experience}xp)
+            </span>
+          )}
+        </h1>
+      </header>
+      <main>
+        <p>
+          This example demonstrates Paima Engine capabilities when used in cooperation with
+          traditional centralized game servers.
+        </p>
+        <ul>
+          <li>
+            Admin functionality (experience gain) - can be modified and stored on chain only from
```

Comment:
Perhaps add message to users know to search and run `post-batcher.mjs`

Reply:
should it be something like `npm run post -- 0xwallet 100` ?

Reply:
Added :+1: